worker/provisioner/provisioner_task.go

```diff
@@ -1334,8 +1351,10 @@ func volumeAttachmentsToAPIServer(attachments []storage.VolumeAttachment) map[st
 	for _, a := range attachments {
 		var planInfo *params.VolumeAttachmentPlanInfo
 		if a.PlanInfo != nil {
-			planInfo.DeviceType = a.PlanInfo.DeviceType
-			planInfo.DeviceAttributes = a.PlanInfo.DeviceAttributes
+			planInfo = &params.VolumeAttachmentPlanInfo{
+				DeviceType:       a.PlanInfo.DeviceType,
+				DeviceAttributes: a.PlanInfo.DeviceAttributes,
+			}
```

Comment:
Hold on a minute, this actually fixes a runtime panic surely?

Reply:
Volume attachment plans (I think) are only used to attach SCSI volumes to OCI instances.

It is possible that no one has added a machine with a volume attachment specified at creation time. I think if one were to attach such a volume to an existing machine, this path would not be traversed.

I could do some A/B testing, but the "popularity" of that provider doesn't justify verifying that this fixes a specific error. I'm happy that it is correct *now*.

Reply:
My worry is that we don't know how this should work, I'm assuming it should work like this, but we're unsure. Can we at least leave a note stating that fact?